File: AdminApi.Sdk/Endpoints/Identities/Types/IdentityDeletionProcessAuditLogEntry.cs
namespace Backbone.AdminApi.Sdk.Endpoints.Identities.Types;

public class IdentityDeletionProcessAuditLogEntry
{
    public required string Id { get; set; }
    public required DateTime CreatedAt { get; set; }
    public required string? OldStatus { get; set; }
    public required string NewStatus { get; set; }
    public required string MessageKey { get; set; }
    public required Dictionary<string, string> AdditionalData { get; set; }
}

File: AdminUi/packages/admin_api_types/lib/src/identities/identity_deletion_process_auditlog_entry.dart
import 'package:json_annotation/json_annotation.dart';

part 'identity_deletion_process_auditlog_entry.g.dart';

@JsonSerializable()
class IdentityDeletionProcessAuditLogEntry {
  final String id;
  final DateTime createdAt;
  final String messageKey;
  final String? oldStatus;
  final String newStatus;
  final Map<String, String> additionalData;

  IdentityDeletionProcessAuditLogEntry({
    required this.id,
    required this.createdAt,
    required this.messageKey,
    required this.newStatus,
    required this.additionalData,
    this.oldStatus,
  });

  factory IdentityDeletionProcessAuditLogEntry.fromJson(dynamic json) => _$IdentityDeletionProcessAuditLogEntryFromJson(json as Map<String, dynamic>);
  Map<String, dynamic> toJson() => _$IdentityDeletionProcessAuditLogEntryToJson(this);
}
